src/components/DogForm.js

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { useParams } from 'react-router-dom';
+import { createDog, updateDog } from '../services/DogList';
+
+export default function DogForm({
+  name,
+  setName,
+  breed,
+  setBreed,
+  bio,
+  setBio,
+  image,
+  setImage,
+  age,
+  setAge,
+}) {
+  const params = useParams();
+  const history = useHistory();
+
+  return (
+    <div>
+      <form
+        onSubmit={(e) => {
+          e.preventDefault();
+          if (params.id) {
+            updateDog(params.id, name, breed, bio, image, age);
+            history.push(`/dog/${params.id}`);
+          } else {
+            window.alert(`${name}!! Welcome to the Alchemy Adoption Center`);
+            createDog(name, breed, bio, image, age);
+            history.push(`/`);
+          }
+        }}
```

Comment:
Remember to call if/else statements above the return in JSX and within their own function.